File: src/main/java/com/logicalgeekboy/logical_zoom/java_manaagers/RenderManager.java
package com.logicalgeekboy.logical_zoom.java_manaagers;

import com.logicalgeekboy.logical_zoom.java_event.EventListener;
import com.logicalgeekboy.logical_zoom.java_event.Priority;
import com.logicalgeekboy.logical_zoom.java_event.impl.Render2DEvent;
import com.logicalgeekboy.logical_zoom.skid;

public class RenderManager {
    public long ms;
    public long lastMs;
    private boolean DEBUG = true;



    public RenderManager() {
        skid.Companion.getEventBus().registerListener(this);
    }

    @EventListener(getPriority= Priority.HIGHEST)
    public void event(Render2DEvent event) {
        long currentTime = System.currentTimeMillis();
        ms = currentTime - lastMs;
        lastMs = currentTime;

        // Cap the MS value to avoid extreme values
        if (ms > 30) {
            ms = 0L;
        } else {
        }
    }

    public float getMs() {
        float result = (float)this.ms * 0.005f;
        return result;
    }
}

File: src/main/kotlin/com/dov/cm/managers/RenderManager.kt
/*
when sex mod for prestige clarinet? - cloovey
*/
package com.dov.cm.managers

import com.dov.cm.event.EventBus
import com.dov.cm.event.EventListener
import com.dov.cm.event.Priority
import com.dov.cm.event.Render2DEvent

class RenderManager(private val eventBus: EventBus) {
    var ms: Long = 0
    var lastMs: Long = 0

    init {
        eventBus.registerListener(this)
    }

    @EventListener(priority = Priority.HIGHEST)
    fun event(event: Render2DEvent?) {
        ms = System.currentTimeMillis() - lastMs
        lastMs = System.currentTimeMillis()
        if (ms > 30) {
            ms = 0L
        }
    }

    fun getMs(): Float {
        return ms.toFloat() * 0.005f
    }

    companion object {
        private var INSTANCE: RenderManager? = null

        fun initialize(eventBus: EventBus) {
            if (INSTANCE == null) {
                INSTANCE = RenderManager(eventBus)
            }
        }

        fun getInstance(): RenderManager {
            return INSTANCE ?: throw IllegalStateException("RenderManager not initialized")
        }
    }
}